packages/legacy-workbench/src/js/angular/guides/steps/core/plugin.js

```diff
@@ -103,6 +110,51 @@ PluginRegistry.add('guide.step', [
             return stepDescription;
         }
     },
+    {
+        guideBlockName: 'copy-text-element',
+        getSteps: (options, services) => {
+            const $translate = services.$translate;
+            const GuideUtils = services.GuideUtils;
+
+            const text = options.text;
+            const code = document.createElement('code');
+            code.innerText = text;
+
+            const copy = document.createElement('button');
+            const copyToEditorButtonClass = 'guide-copy-to-editor-query-button';
+            copy.className = `btn btn-sm btn-secondary ${copyToEditorButtonClass}`;
+            copy.innerText = $translate.instant('guide.step_plugin.execute-sparql-query.copy-to-editor.button');
+
+
+            let stepHTMLElement;
+            let copyToEditorListener;
+
+            return [
+                {
+                    guideBlockName: 'input-element',
+                    options: {
+                        ...options,
+                        content: 'guide.step_plugin.core-steps.copy-text-element.content',
+                        textAsHtmlCodeElement: '<div class="shepherd-code">' + code.outerHTML + copy.outerHTML + '</div>',
+                        extraContent: options.queryExtraContent,
+                        show: (_guide) => () => {
+                            const elementSelector = GuideUtils.getElementSelector(options.elementSelector);
+                            const inputElement = document.querySelector(elementSelector);
+
+                            copyToEditorListener = createCopyToInputListener(inputElement, text);
```

Comment:
The listener should be initialized in the same scope where let copyToEditorListener; is declared. This avoids recreating the listener every time the step shows.